src/openvic-simulation/map/Province.cpp

```diff
@@ -137,6 +137,18 @@ bool Province::add_adjacency(Province const* province, distance_t distance, flag
 	return true;
 }
 
+fvec2_t Province::get_unit_position() const {
+	return positions.unit.value_or(positions.center);
+}
+
+Province::distance_t Province::calculate_distance_to(Province const* province) const {
+	const fvec2_t my_unit_position = get_unit_position();
+	const fvec2_t other_unit_position = province->get_unit_position();
+	const fvec2_t distance_vector = other_unit_position - my_unit_position;
+	const fixed_point_t distance = distance_vector.length_squared();
```

Comment:
This is distance squared. I need to math.sqrt it. We just need a sqrt for fixed point types.
https://github.com/MikeLankamp/fpm/blob/master/include/fpm/math.hpp is an option

Reply:
The square root function you linked is based on [this algorithm](https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Binary_numeral_system_.28base_2.29), looks like a good candidate 👍
For now though let's leave this as the square of the length, so we can get this merged and I can continue my work on `additional-adjacencies` on top of it.